File: mybatis-test/src/test/java/com/example/mybatistest/UserRepoTest.java
package com.example.mybatistest;

import com.github.pagehelper.PageInfo;
import org.junit.jupiter.api.Test;
import org.mybatis.spring.boot.test.autoconfigure.MybatisTest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Import;
import org.springframework.data.domain.PageRequest;
import org.springframework.test.context.jdbc.Sql;

import java.time.LocalDateTime;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNotNull;

@Import(PageHelperTestConfig.class)
@MybatisTest
class UserRepoTest {

    @Autowired
    private UserRepo userRepo;


    @Sql(statements = {
            "insert into user(id, name, create_time) values ('1', '张三', '2021-06-21')",
            "insert into user(id, name, create_time) values ('2', '李四', '2021-06-23')",
    })
    @Test
    void findAll() {
        assertNotNull(userRepo);
        List<User> all = userRepo.findAll(LocalDateTime.MIN, LocalDateTime.MAX);
        assertEquals(2, all.size());
    }

    @Sql(statements = {
            "insert into user(id, name, create_time) values ('1', '张三', '2021-06-21')",
            "insert into user(id, name, create_time) values ('2', '李四', '2021-06-23')",
    })
    @Test
    void findUsers() {
        assertNotNull(userRepo);
        UserQuery userQuery = new UserQuery().setEndTime(LocalDateTime.of(2021, 6, 22, 0, 0));
        List<User> all = userRepo.findUsers(userQuery);
        assertEquals(1, all.size());
    }


    @Sql(statements = {
            "insert into user(id, name, create_time) values ('1', '张三', '2021-06-21')",
            "insert into user(id, name, create_time) values ('2', '李四', '2021-06-23')",
    })
    @Test
    void findPage() {
        UserQuery userQuery = new UserQuery();
        PageInfo<User> page = userRepo.findPage(userQuery, PageRequest.of(0, 1));
        assertEquals(2, page.getTotal());
        assertEquals(1, page.getList().size());
    }
}

File: spring-test/src/test/kotlin/cn/zzk/springtest/UserRepoTest.kt
package cn.zzk.springtest

import org.junit.jupiter.api.Assertions.assertEquals
import org.junit.jupiter.api.Assertions.assertNotNull
import org.junit.jupiter.api.DisplayName
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest
import org.springframework.boot.test.autoconfigure.orm.jpa.TestEntityManager
import org.springframework.test.context.jdbc.Sql

@DataJpaTest
class UserRepoTest {

    @Autowired
    lateinit var userRepo: UserRepo

    @Autowired
    lateinit var entityManager: TestEntityManager

    @Test
    fun notNull() {
        entityManager.merge(User(name = "张三", age = 18))
        assertNotNull(userRepo)
        assertEquals(1, userRepo.count())
    }

    @Test
    @Sql(
        statements = [
            "insert into user (id,name,age) values ('1','张三',18)"
        ]
    )
    fun jdbcTest() {
        assertEquals(1, userRepo.count())
    }

    @Test
    @Sql(
        statements = [
            "insert into organ_unit (id,name) values ('1','温州市局')",
            "insert into user (id,name,age,organ_unit_id) values ('1','张三',18,'1')",
        ]
    )
    @DisplayName("dto 查询")
    fun findUserDTOs() {
        val users = userRepo.findAllUserDTOs()
        assertEquals(1, users.size)
        assertEquals("1",users[0].id)
        assertEquals("温州市局",users[0].unitName)
    }

}